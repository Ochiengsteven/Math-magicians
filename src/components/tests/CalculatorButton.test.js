import React from 'react';
import { render } from '@testing-library/react';
import CalculatorButton from '../calculatorButton'; // Adjust the import path if needed

describe('CalculatorButton', () => {
  it('should render correctly', () => {
    const { container } = render(<CalculatorButton label="1" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with double width', () => {
    const { container } = render(<CalculatorButton label="0" isDoubleWidth />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as last button in row', () => {
    const { container } = render(<CalculatorButton label="+" isLastInRow />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
