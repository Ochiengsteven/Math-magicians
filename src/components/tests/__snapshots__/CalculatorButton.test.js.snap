// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton should render as last button in row 1`] = `
<button
  class="calc-button  operation-button"
  type="button"
>
  +
</button>
`;

exports[`CalculatorButton should render correctly 1`] = `
<button
  class="calc-button  "
  type="button"
>
  1
</button>
`;

exports[`CalculatorButton should render with double width 1`] = `
<button
  class="calc-button double-width "
  type="button"
>
  0
</button>
`;
